File: backend/app/schemas/paper.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel

from app.models.paper import PaperStatus


class PaperRead(BaseModel):
    id: UUID
    title: str | None = None
    authors: str | None = None
    year: int | None = None
    abstract: str | None = None
    filename: str
    status: PaperStatus
    error_message: str | None = None
    uploaded_at: datetime

    model_config = {"from_attributes": True}


class PaperChunkRead(BaseModel):
    id: UUID
    paper_id: UUID
    chunk_index: int
    page_number: int | None = None
    text: str
    token_count: int | None = None

    model_config = {"from_attributes": True}

File: frontend/src/types/paper.ts
export type PaperStatus = "queued" | "processing" | "ready" | "failed";

export type Paper = {
  id: string;
  title: string | null;
  authors: string | null;
  year: number | null;
  abstract: string | null;
  summary: string | null;
  profile: Record<string, unknown> | null;
  filename: string;
  status: PaperStatus;
  error_message: string | null;
  uploaded_at: string;
};